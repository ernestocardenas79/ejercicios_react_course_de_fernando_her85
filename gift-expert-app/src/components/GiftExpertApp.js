import React, {useState } from 'react';
import { AddCategory } from './AddCategory';
import { GiftGrid } from './GiftGrid';

export const GiftExpertApp = ({defaultCategories = []})=> {

    const [categories, setCategories] = useState(defaultCategories );
    
    return (
        <>
            <h2>GiftExpertApp</h2>   
            <AddCategory 
                setCategories={setCategories}
                /> 
            <hr/>
            <ol>
                {
                    categories.map(category=>(
                        <GiftGrid 
                            category={category} 
                            key={category}
                          />
                    ))
                }
            </ol>  
        </>
    )
}

export default GiftExpertApp
