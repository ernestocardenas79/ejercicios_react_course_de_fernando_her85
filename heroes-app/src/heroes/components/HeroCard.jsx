import { Link } from "react-router-dom"

export const HeroCard = ({id, superhero,alter_ego, characters, first_appearance})=>{
    
    return (
        <div className="col animate__animated animate__fadeIn">
            <div className="card">
                <div className="row no-glutters">
                    <div className="col-4">
<img className="card-img" src={`assets/heroes/${id}.jpg`} alt={superhero} />
                    </div>
                    <div className="col-8">
<div className="card-body">
                <h5 className="card-title">{superhero}</h5>
                <h6 className="card-text">{alter_ego}</h6>
                {
                    (alter_ego !== characters) && (<p>{characters}</p>)
                }

                <p className="card-text">
                    <small>{first_appearance}</small>
                </p>

                <Link to={`/hero/${id}`}>
                    Mas...
                </Link>
                </div>
                    </div>
                    
                </div>
            </div>
        </div>
    )
}