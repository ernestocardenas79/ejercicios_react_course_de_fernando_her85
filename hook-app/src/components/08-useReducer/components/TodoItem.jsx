
export const TodoItem = ({todo, handleToggle, handleDelete, index}) => {
  return (
    <li className="list-group-item">
      <p
        className={`${todo.done && "complete"}`}
        onClick={() => handleToggle(todo.id)}
      >
        {index + 1}. {todo.desc}
      </p>
      <button
        className="btn btn-danger"
        type="button"
        onClick={() => handleDelete(todo.id)}
      >
        Borrar
      </button>
    </li>
  );
};
