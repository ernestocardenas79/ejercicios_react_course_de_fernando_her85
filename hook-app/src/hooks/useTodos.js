import { useEffect, useReducer } from "react";
import { todoReducer } from "../components/08-useReducer/todoReducer";

const init = () => JSON.parse(localStorage.getItem("todos")) || [];

export const useTodos = ()=>{
    const [todos, todoDispatch] = useReducer(todoReducer, [], init);

    useEffect(() => {
        localStorage.setItem("todos", JSON.stringify(todos));
      }, [todos]);


      const newTodoHandler = (newTodo) => {
        todoDispatch({ type: "add", payload: newTodo });
      };
    
      const deleteTodoHandler = (todoId) => {
        const action = {
          type: "delete",
          payload: todoId,
        };
    
        todoDispatch(action);
      };
    
      const toggleTodoHandler = (todoId) => {
        todoDispatch({
          type: "toggle",
          payload: todoId,
        });
      };

    return {
        todos,
        deleteTodoHandler,
        toggleTodoHandler,
        newTodoHandler,
        todosCount: todos.length,
        pedingTodosCount: todos.filter(todo=>!todo.done).length
    }
}