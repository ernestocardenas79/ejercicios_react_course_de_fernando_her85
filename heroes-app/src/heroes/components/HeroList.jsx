import { useMemo } from "react";
import { getHeroesByPublisher } from "../helpers";
import { HeroCard } from "./HeroCard";

export const HeroesList= ({publisher})=>{
    const heroesList = useMemo(()=>getHeroesByPublisher(publisher),[publisher]);
    
    return (
        <div className="row rows-cols-1 row-cols-md-3 g-3">
             {heroesList.map(h=> (<HeroCard  key={h.id} {...h}></HeroCard>))}

        </div>
    )
}