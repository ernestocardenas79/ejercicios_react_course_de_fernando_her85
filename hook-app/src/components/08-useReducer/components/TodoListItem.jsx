import { TodoItem } from "./TodoItem";

export const TodoListItem = ({ todos, handleToggle, handleDelete }) => {
  return (
    <ul className="list-group list-group-flush">
      {todos.map((todo, i) => (
        <TodoItem
          key={todo.id}
          todo={todo}
          handleToggle={handleToggle}
          handleDelete={handleDelete}
          index={i}
        />
      ))}
    </ul>
  );
};
