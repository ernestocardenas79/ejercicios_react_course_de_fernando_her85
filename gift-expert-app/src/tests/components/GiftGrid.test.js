import React from 'react'; 
import { shallow } from 'enzyme';
import { GiftGrid } from '../../components/GiftGrid';
import '@testing-library/jest-dom';

import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Prueba de componente <GiftGrid />', () => {
    const category ='Mazinger';
    
    test('Debe de crearce el componente',()=>{

        useFetchGifs.mockReturnValue({
            data:[],
            loading:true
        });

        const wrapper = shallow(<GiftGrid category={category}/>);

        expect(wrapper).toMatchSnapshot();
    })

    test('debe de mostrar items cuando se cargan imÃ¡genes useFechtGifs ', () => {
        const gifs=[{
            id:'ABC',
            url:'https://cualquiercosa/imagen/cosa.jpg',
            title: 'Cualquier cosa'
        }];
        
        useFetchGifs.mockReturnValue({
            data:gifs,
            loading:false
        });

        const wrapper = shallow(<GiftGrid category={category}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GiftGridItem').length).toBe(gifs.length);
    })    
})
