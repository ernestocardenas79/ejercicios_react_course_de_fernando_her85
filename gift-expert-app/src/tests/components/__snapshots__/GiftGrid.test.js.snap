// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de componente <GiftGrid /> Debe de crearce el componente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Mazinger
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid "
  />
</Fragment>
`;

exports[`Prueba de componente <GiftGrid /> debe de mostrar items cuando se cargan im√°genes useFechtGifs  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Mazinger
  </h3>
  <div
    className="card-grid "
  >
    <GiftGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://cualquiercosa/imagen/cosa.jpg"
    />
  </div>
</Fragment>
`;
